components/map/index.jsx

```diff
@@ -0,0 +1,162 @@
+	 * Title for the Map component.
+	 */
+	title: PropTypes.string,
+	/**
+	 * Accepts a single Map location
+	 */
+	location: PropTypes.string,
+	/**
+	 * Accepts Map locations as an array of Objects
+	 */
+	locations: PropTypes.arrayOf(PropTypes.object),
+	/**
+	 * Accepts a Google Map API Key that will be used for showing the map
+	 */
+	googleAPIKey: PropTypes.string.isRequired,
+	/**
+	 * Accepts a node that will be shown at the footer of the modal containing the map. Only for `modal` variant
+	 */
+	modalFooter: PropTypes.node,
+	/**
+	 * Selects the variant of Map component
+	 */
+	variant: PropTypes.oneOf(['standalone', 'modal']),
+};
+
+const defaultProps = {
+	isModalOpen: true,
+	variant: 'standalone',
+};
+
+/**
+ * A map component is used to find a location
+ */
+class Map extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			selected: 0,
+		};
+		this.generatedId = shortid.generate();
+	}
+
+	/**
+	 * Get the Map's HTML id. Generate a new one if no ID present.
+	 */
+	getId() {
+		return this.props.id || this.generatedId;
+	}
+
+	render() {
+		const mapContainer = (
+			<div
+				id={this.getId()}
+				className={classNames(
+					`slds-grid`,
+					{'slds-has-coordinates' : this.props.locations},
+					this.props.className
+				)}
+			>
+				<div className="slds-map_container">
+					<div className="slds-map">
+						<iframe
+							id={`${this.getId()}-google-map`}
+							title="Google Maps iframe"
+							src={`https://www.google.com/maps/embed/v1/place?key=${
+								this.props.googleAPIKey
+							}&q=${ this.props.locations ?
+								encodeURIComponent(this.props.locations[this.state.selected].address)
+								: encodeURIComponent(this.props.location)
+							}`}
+						/>
+					</div>
+				</div>
+				{ this.props.locations ? (
+				<div className="slds-coordinates">
+					<div className="slds-coordinates__header">
+						<h2 className="slds-coordinates__title">{`${this.props.title} (${
+							this.props.locations.length
+						})`}</h2>
+					</div>
+					<ul className="slds-coordinates__list">
+						{this.props.locations.map((location, i) => (
+							<li key={location.name} className="slds-coordinates__item">
+								<span className="slds-assistive-text" aria-live="polite" />
+								<button
+									type="button"
+									onClick={() => this.setState({ selected: i })}
+									className="slds-coordinates__item-action slds-button_reset slds-media"
+									aria-pressed={this.state.selected === i}
+								>
+									<span className="slds-media__figure">
+										<Icon category="standard" name="account" />
+									</span>
+									<span className="slds-media__body">
+										<span className="slds-text-link">{location.name}</span>
+										<span>{location.address}</span>
+									</span>
+								</button>
+							</li>
+						))}
+					</ul>
+				</div>) : null }
+			</div>
+		);
+
+		return this.props.variant === 'standalone' ? (
+			mapContainer
+		) : (
+			<Modal
```

Comment:
Consumers are going to need to be able to control all the modal props.

We have three options to allow that. We can pass a `Modal` into a `modal` prop of `Map` or we can just do an example of a `Modal` with a `Map` as content. I'm leaning toward doing an example of a `Modal` with a `Map` as content in the example code, however, I'm open to another option which is create a sub component something like `MapModal` within the `map` folder. it would take all the Modal props by default default. See https://github.com/salesforce/design-system-react/blob/master/components/global-navigation-bar/dropdown.jsx#L25 if you'd like to do that.